docs/user-guide/manage-branches.md

```diff
@@ -1,6 +1,30 @@
 # Manage Branches
 
-This page describes how to manage branches in the created component, whether it is an application, library, autotest or infrastructure.
+This page describes how to manage branches in the created component, whether it is an application, library, autotest or infrastructure. It also briefly explains two approaches of managing custom pipelines for codebases.
+
+## Custom Pipeline Management
+
+There are two approaches to define a codebases with custom Tekton pipelines:
+
+  * **Using custom build tool and framework**
+  * **Using codebase branch settings**
+
+### Custom Build Tool/Framework Approach
+
+This approach involves [adding your application](../use-cases/tekton-custom-pipelines.md) with custom language type and framework.
+
+Follow this approach if you need to define multiple similar applications with identical settings, as this approach would be more beneficial.
+
+###  Codebase Branch Settings Add-Ons Approach
+
+This approach involves creating a component with custom Tekton pipelines in your [Add-Ons repository](../operator-guide/add-ons-overview.md) and selecting it in the codebase branch settings.
+
+Unlike the [approach with custom build tool and framework](../use-cases/tekton-custom-pipelines.md), the new one offers two main advantages:
+
+  * **Simplified management**: The new approach eliminates the need for strict naming conventions for resources. In the first approach, it is essential to create a pipeline name using the heavy pattern: `git_server_name-build_tool-framework-codebase_type-pipeline_type-codebase_versioning_type`. In this approach, there is no need to follow any patterns when creating pipelines.
```

Comment:
Maybe we need to migrate description with resource name pattern to appropriate section?
git_server_name-build_tool-framework-codebase_type-pipeline_type-codebase_versioning_type looks like must be in ### Custom Build Tool/Framework Approach?